src/recordlinker/routes/algorithm_router.py

```diff
@@ -58,11 +59,15 @@ def create_algorithm(
     :returns: The created algorithm
     """
     try:
-        service.load_algorithm(session, data)
+        service.load_algorithm(session, data, commit=True)
     except ValueError as exc:
         raise fastapi.HTTPException(
             status_code=fastapi.status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(exc)
         )
+    except sqlexc.IntegrityError as exc:
+        raise fastapi.HTTPException(
```

Comment:
@m-goggins @bamader this is unrelated to the GH issue, but a bug I found while testing yesterday.  If a user tries to create an algorithm with a non-unique label, a server error is generated rather than a 422.  I hate seeing 500s in the code, so I try to fix those as fast as possible, and thus put it in here.

I added a test case for this in test_algorithm_router.py and verified locally that this block was working as expected. I'm not sure why codedov is saying otherwise.

Reply:
Oh wonky. Thanks for letting us know! I was gonna flag it as needing coverage but if you're good locally then yeah I don't know what's up with codecov. Also good catch on that error--I hate debugging and seeing 500s so anything to clean them up is a good change in my eyes.